krakenapi.go

```diff
@@ -60,8 +60,13 @@ type KrakenApi struct {
 }
 
 // New creates a new Kraken API client
-func New(key, secret string) *KrakenApi {
-	client := &http.Client{}
+func New(key, secret string, httpClient ...*http.Client) *KrakenApi {
```

Comment:
Can you created a new method instead of expanding the old one?

Like `krakenapi.NewWithClient(key, secret, client)`. It is cleaner in my opinion.